lib/cpp/test/OpenSSLManualInitTest.cpp

```diff
@@ -67,6 +67,22 @@ void test_openssl_availability() {
   openssl_cleanup();
 }
 
```

Comment:
Just an idea but BOOST_TEST_MAIN or (even better) BOOST_TEST_MODULE work instead of these `main`s ?

I'm fine with the current approach, though.
We have a bug report for boost test dynamic linking and this certainly solves it and it's great.